src/updatedownloader.h

```diff
@@ -28,42 +28,111 @@
 
 #include "threads.h"
 #include "appcast.h"
+#include "download.h"
+#include "ui.h"
 
 #include <string>
+#include <time.h>
 
 namespace winsparkle
 {
-
-/**
-    This class performs application update.
-
-    If an update is found, then UpdateChecker initializes the GUI thread
-    and shows information about available update to the user.
- */
-class UpdateDownloader : public Thread
-{
-public:
-    /// Creates updater thread.
-    UpdateDownloader(const Appcast& appcast);
+    // helper functions
+    std::wstring GetUniqueTempDirectoryPrefix();
+    std::wstring CreateUniqueTempDirectory();
 
     /**
-        Perform any necessary cleanup after previous updates.
+        This class performs application update.
 
-        Should be called on launch to get rid of leftover junk from previous
-        updates, such as the installer files. Call it as soon as possible,
-        before using other WinSparkle functionality.
+        If an update is found, then UpdateChecker initializes the GUI thread
+        and shows information about available update to the user.
      */
-    static void CleanLeftovers();
+    class UpdateDownloader : public Thread
+    {
+    public:
+        /// Creates updater thread.
+        UpdateDownloader(const Appcast& appcast);
+
+        /**
+            Perform any necessary cleanup after previous updates.
+
+            Should be called on launch to get rid of leftover junk from previous
+            updates, such as the installer files. Call it as soon as possible,
+            before using other WinSparkle functionality.
+         */
+        static void CleanLeftovers();
+
+    protected:
+        // Thread methods:
+        virtual void Run();
+        virtual bool IsJoinable() const { return true; }
+
+    private:
+        Appcast m_appcast;
+    };
+
+
+    struct UpdateDownloadSink : public IDownloadSink
```

Comment:
Why is this implementation (definition) moving into the header file? Shouldn't it stay in the cpp file and just have a declaration here in the header file if it needs to be exposed?